tests/unit_tests/legacy_tests.py

```diff
@@ -0,0 +1,134 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, use-implicit-booleaness-not-comparison
+
+import copy
+from typing import Any
+
+from superset.legacy import update_time_range
+from tests.unit_tests.conftest import with_feature_flags
+
+
+def test_update_time_range_since_until() -> None:
+    """
+    Tests for the old `since` and `until` parameters.
+    """
+    form_data: dict[str, Any]
+
+    form_data = {}
+    update_time_range(form_data)
+    assert form_data == {}
+
+    form_data = {"since": "yesterday"}
+    update_time_range(form_data)
+    assert form_data == {"time_range": "yesterday : "}
+
+    form_data = {"until": "tomorrow"}
+    update_time_range(form_data)
+    assert form_data == {"time_range": " : tomorrow"}
+
+    form_data = {"since": "yesterday", "until": "tomorrow"}
+    update_time_range(form_data)
+    assert form_data == {"time_range": "yesterday : tomorrow"}
+
+
+@with_feature_flags(GENERIC_CHART_AXES=False)
+def test_update_time_range_granularity_sqla_no_feature_flag() -> None:
+    """
+    Tests for the unfiltered `granularity_sqla` when `GENERIC_CHART_AXES` is off.
+    """
+    original_form_data = {
```

Comment:
We could define `original_form_data` with only the attributes we care about when testing this function. It seems we only need:
- since
- until
- time_range
- granularity_sqla
- adhoc_filters

You can also extract `original_form_data` to a variable and reuse it in the tests as they are identical.

Reply:
Ah, good point in creating a variable, will do!

I like leaving the full `form_data`, even if it has fields that aren't needed, because it's a real `form_data` from a real chart, so it brings in some additional context. Also, in the future if someone needs to add tests that rely on additional fields, they don't have to figure it out, or risk adding fields that make the `form_data` inconsistent.

Reply:
Got it. Makes sense!